snap/quota/resources.go

```diff
@@ -189,7 +241,7 @@ func (qr *Resources) Validate() error {
 func (qr *Resources) ValidateChange(newLimits Resources) error {
 	// Check that the memory limit is not being decreased
 	if newLimits.Memory != nil {
-		if newLimits.Memory.Limit == 0 {
+		if qr.Memory != nil && newLimits.Memory.Limit == 0 {
```

Comment:
These changes are a bit strange, because they are part of a commit that says only "Fix unit tests"; but unit tests are not currently broken, as far as I can tell. If you still think that these changes are needed, I'd suggest proposing them as a separate PR, which a commit message that explains why they are needed.

Reply:
I think it was a bit confusing here, I fixed unit tests because of an error I introduced myself in earlier commits in this PR. I had to change some of these checks because some checks now only apply if the there is an pre-existing limit set.